# 16 · GSE232381 · bulk_RNA_seq · module-trait associations

Reads `expression.rds`, `modules.rds`, `metadata.rds`, and the GSE65391 end product `module_genes.csv`.
Writes the end products `module_traits.csv` and `clinical_dictionary.csv` in `data/run_artifacts/GSE232381/`.

**Trait:** `nephritis_active`, 1 for active and 0 for inactive lupus nephritis (12 and 13 samples).

**Eigengenes**, two families:
1. **array modules**: the projected scores of notebook 15b (GSE65391 module weights applied to this
   study's standardised genes); these are the ones compared across studies;
2. **own modules**: the modules of notebook 14.

**Test.** Pearson (point-biserial) r between eigengene and `nephritis_active`, p from `cor.test`, BH q
within each family. n = 25. **A finding** is q < 0.05.

## Load the two sets of module scores and the trait

In [1]:
if (!requireNamespace("WGCNA", quietly = TRUE)) install.packages("WGCNA", repos = "https://cloud.r-project.org")
suppressMessages(library(WGCNA))
source("../src/paths.R")
x    <- readRDS(art("GSE232381", "expression.rds"))
own  <- readRDS(art("GSE232381", "modules.rds"))
meta <- readRDS(art("GSE232381", "metadata.rds"))
datExpr <- t(x$E)
ME_array <- readRDS(art("GSE232381", "projected_scores.rds"))$scores[rownames(datExpr), ]
colnames(ME_array) <- paste0("ME", colnames(ME_array))
ME_own   <- own$MEs[, colnames(own$MEs) != "MEgrey"]
trait <- meta[rownames(datExpr), "ln_active"]
table(nephritis_active = trait)

nephritis_active
 0  1 
13 12 

**Explanation**
- `ME_array`: the array modules projected into these samples (notebook 15b); `paste0("ME", ...)` names the
  columns like WGCNA eigengenes.
- `ME_own`: this study's own module eigengenes (notebook 14), without grey.
- `trait`: 1 for active nephritis, 0 for inactive.

## Correlate each module with nephritis activity

In [2]:
test_family <- function(ME, family) {
  out <- do.call(rbind, lapply(colnames(ME), function(me) {
    ct <- cor.test(ME[, me], trait)
    data.frame(family = family, module = sub("^ME", "", me), trait = "nephritis_active", n = length(trait),
               r = unname(ct$estimate), p = ct$p.value)
  }))
  out$q <- p.adjust(out$p, "BH"); out[order(out$p), ]
}
res_array <- test_family(ME_array, "array module")
res_own   <- test_family(ME_own, "own module")
format(res_array, digits = 3)
head(format(res_own, digits = 3), 10)
c(array_modules_q_below_0.05 = sum(res_array$q < 0.05), own_modules_q_below_0.05 = sum(res_own$q < 0.05))

,family,module,trait,n,r,p,q
,<I<chr>>,<I<chr>>,<I<chr>>,<I<chr>>,<I<chr>>,<I<chr>>,<I<chr>>
10,array module,red,nephritis_active,25,-0.4340,0.0302,0.196
9,array module,purple,nephritis_active,25,-0.4124,0.0405,0.196
3,array module,brown,nephritis_active,25,-0.4095,0.0421,0.196
6,array module,greenyellow,nephritis_active,25,-0.3319,0.1050,0.345
14,array module,yellow,nephritis_active,25,0.3165,0.1232,0.345
11,array module,salmon,nephritis_active,25,-0.2740,0.1851,0.432
1,array module,black,nephritis_active,25,-0.2483,0.2315,0.463
4,array module,cyan,nephritis_active,25,-0.1912,0.3598,0.486
2,array module,blue,nephritis_active,25,-0.1896,0.3639,0.486


,family,module,trait,n,r,p,q
,<I<chr>>,<I<chr>>,<I<chr>>,<I<chr>>,<I<chr>>,<I<chr>>,<I<chr>>
13,own module,brown,nephritis_active,25,0.43329,0.0305,0.311
3,own module,turquoise,nephritis_active,25,-0.41637,0.0384,0.311
8,own module,darkgreen,nephritis_active,25,-0.41558,0.0388,0.311
12,own module,royalblue,nephritis_active,25,0.35853,0.0784,0.324
23,own module,blue,nephritis_active,25,0.35106,0.0853,0.324
2,own module,tan,nephritis_active,25,-0.32516,0.1127,0.324
6,own module,grey60,nephritis_active,25,-0.32388,0.1142,0.324
5,own module,yellow,nephritis_active,25,-0.32056,0.1182,0.324
10,own module,darkgrey,nephritis_active,25,0.31795,0.1214,0.324


array_modules_q_below_0.05   own_modules_q_below_0.05 
                         0                          0

**Explanation**
- `test_family(ME, family)` tests every module of one set:
  - `cor.test(score, trait)` gives the Pearson correlation and its p value. With a 0/1 trait, this is the
    point-biserial correlation, equivalent to a t test of the score between the two groups.
  - `p.adjust(out$p, "BH")` gives Benjamini-Hochberg q values within the set (see notebook 06b).
- The two sets are tested separately, each with its own correction.

**Result.** No module, array or own, is associated with nephritis activity at q < 0.05. The largest
array-module effects are red (r = -0.43, p = 0.030), purple (r = -0.41, p = 0.041) and brown (r = -0.41,
p = 0.042); the largest own-module effect is brown (r = 0.43, p = 0.031). With 25 samples, |r| must exceed
0.40 to reach p < 0.05 before correction.

## Save the end products

In [3]:
res <- rbind(res_array, res_own)
res$fisher_z <- atanh(res$r); res$se <- 1 / sqrt(res$n - 3)
write.csv(res, art("GSE232381", "module_traits.csv"), row.names = FALSE)
write.csv(data.frame(study = "GSE232381", trait = "nephritis_active", recorded = sum(!is.na(trait)), binary = TRUE),
          art("GSE232381", "clinical_dictionary.csv"), row.names = FALSE)

**Explanation**
- `atanh(r)` is Fisher's z transform of r; `1 / sqrt(n - 3)` its standard error. These let the federation
  notebook combine correlations across studies.
- `module_traits.csv` and `clinical_dictionary.csv` are the end products of this study.

## Comparison with Chen et al. 2024 (*Heliyon* 10:e32303)

- **C0** (sample discrepancy) and **C1** (definition of activity) are stated in notebook 11.
- **C3.** By flow cytometry, active nephritis had lower NK and memory CD4+ T-cell percentages and higher
  naive CD4+ and CD8+ T-cell percentages (Results 2.1, Fig. 1C, Table 2). Test here: the module score
  closest to these cell types: turquoise, which holds the lymphoid, B-cell and NK genes of GSE65391 (notebook
  05); there is no separate NK module at this power. This compares module scores with cell counts, so it can agree only in direction.
- **C4.** "Active LN increased IFN-gamma activity compared to inactive LN" (Results 2.5, Fig. 8A). The black
  module (interferon) is a type I interferon module, a different pathway; the test is reported, but it is not the
  paper's measure.
- **C2** (the paper's named differentially expressed genes) is a separate notebook, because it needs a
  count-based test.

In [4]:
res_array[res_array$module %in% c("turquoise", "black"), c("module", "n", "r", "p", "q")]   # turquoise: lymphoid and NK genes; black: interferon (notebook 05)

,module,n,r,p,q
,<chr>,<int>,<dbl>,<dbl>,<dbl>
1,black,25,-0.24825905,0.2314662,0.4629323
13,turquoise,25,-0.01054326,0.9601075,0.9601075


**Explanation**
- The rows of the two array modules named in claims C3 and C4.

**Result.**
- **C3.** Turquoise, which holds the NK and lymphoid genes, does not differ with activity (r = -0.01,
  p = 0.96), and the module is not specific to NK cells or to naive or memory T cells.
- **C4, not the same measure.** The type I interferon module (black) does not differ with activity
  (r = -0.25, p = 0.23). The paper's claim concerns interferon-gamma (type II); it is neither supported nor
  contradicted here.